src/CoreBundle/private/library/classes/TCMSFields/TCMSField.class.php

```diff
@@ -659,28 +680,51 @@ public function CreateFieldIndex($returnDDL = false)
             $oFieldType = new TCMSRecord('cms_field_type', $cmsFieldTypeId);
         }
 
-        if ('index' !== $oFieldType->sqlData['indextype'] && 'unique' !== $oFieldType->sqlData['indextype']) {
+        $indexType = strtoupper($oFieldType->sqlData['indextype']);
+
+        if ('INDEX' !== $indexType && 'UNIQUE' !== $indexType) {
             return $returnDDL ? '' : null;
         }
 
-        if ('index' === $oFieldType->sqlData['indextype']) {
-            $indexType = 'INDEX';
-        } elseif ('unique' === $oFieldType->sqlData['indextype']) {
-            $indexType = 'UNIQUE';
+        if (true === $returnDDL) {
```

Comment:
Could be moved only to the return. The `getIndexQuery` is done in any case. Or it should be?

Reply:
Not really. getIndexQuery is also called in createIndex, so we don't want to call it twice. This database description export feature is in our way here. Or do i miss something?

Reply:
I am mainly wondering if true === $returnDDL means that no change should be done (only the query returned).

Reply:
This is the intension here. Do nothing, just return the SQL that would run.